### DISCUSSION TITLE
wanna  use amd gpu  how to  modify the code

### DISCUSSION THREAD

**Discussion by bibiwhat** [ultralytics/ultralytics#3442] [Q&A]
i just try sd1.5 with rocm5.4.2＋pytorch 2.0.1 it it much quick than only using CPU

**Comment by glenn-jocher** [comment]
@bibiwhat hello! It's great to hear that you're interested in using an AMD GPU with YOLOv8. To utilize an AMD GPU, you'll need to ensure that your environment is set up correctly with ROCm and a compatible version of PyTorch that supports AMD GPUs.

Since YOLOv8 is built on top of PyTorch, as long as you have the appropriate PyTorch version installed with ROCm support, YOLOv8 should be able to leverage your AMD GPU without any modifications to the code. Make sure that PyTorch is recognizing your AMD GPU by running `torch.cuda.is_available()` in your Python environment. If it returns `True`, you should be good to go.

If you encounter any issues or have specific questions, feel free to open an issue on the repo, and we'll be happy to help you out. For more detailed instructions on the setup and usage of YOLOv8, please refer to our documentation at https://docs.ultralytics.com. Happy detecting! 😊🚀
